File: ProducerConsumer/src/Solution/Consumer.java
package Solution;

public class Consumer extends Thread{
    private int count = 0;
    Monitor monitor;
    ProducerConsumer producerConsumer;
    Consumer(Monitor monitor, ProducerConsumer producerConsumer){
        this.monitor = monitor;
        this.producerConsumer = producerConsumer;
    }
    @Override
    public void run() {
        while (!producerConsumer.getThreadFlag()){
            if(count < 3) {
                count++;
                monitor.get();
            }else {
                try {
                    Thread.sleep(1000);
                }catch (InterruptedException e) {
                }
                count = 0;
            }
        }
        System.out.println("Solution.Consumer закончил работу: " + currentThread().getId());
    }
}

File: task1/src/main/kotlin/spbu/mm/parallel/programming/Consumer.kt
package spbu.mm.parallel.programming

class Consumer<T>(
    private val taskContainer: TaskContainer<T>,
    private val consume: (T) -> Unit,
    private val sleepTimer: Long = SLEEP_TIME,
    order: Int = 0
) : Thread(order.toString()) {
    override fun run() {
        try {
            while (!isInterrupted) {
                val task = taskContainer.getFirstOrNull()
                if (task != null) consume(task)
                sleep(sleepTimer)
            }
        } catch (ex: InterruptedException) {
            println("Consumer №$name was interrupted")
        }
    }
}
